cdap-ui/app/directives/complex-schema/complex-schema.js

```diff
@@ -108,13 +115,29 @@ function ComplexSchemaController (avsc, SCHEMA_TYPES, $scope, uuid, $timeout) {
       return obj;
     });
 
-    let obj = {
-      type: 'record',
-      name: recordName || uuid.v4(),
-      fields: outputFields
-    };
+    if (outputFields.length > 0) {
+      let obj = {
+        type: 'record',
+        name: recordName || 'a' + uuid.v4().split('-').join(''),
+        fields: outputFields
+      };
+
+      // Validate
+      try {
+        avsc.parse(obj);
+      } catch (e) {
+        let err = '' + e;
+        err = err.split(':');
```

Comment:
parsing the object and recording the error seems to be common across all schema types. Is it possible to add this in some `SchemaHelper` factory?